pages/connect-wallets.tsx

```diff
@@ -0,0 +1,70 @@
+import type { NextPage } from 'next';
+import WalletConnector from '../components/WalletConnector';
+import { useContext } from 'react';
+import { useRouter } from 'next/router';
+import { WalletAddressContext } from '../contexts/WalletAddressContext';
+import { toast } from 'react-toastify';
+import { L16_CHAIN_ID } from '../constants';
+
+const ConnectWallets: NextPage = () => {
+  const router = useRouter();
+  const { setWalletAddress } = useContext(WalletAddressContext);
+
+  const verifyChainId = () => {
+    const chainId = parseInt(window.ethereum.networkVersion);
+    if (chainId && window.ethereum.networkVersion != L16_CHAIN_ID) {
+      toast('Please switch to L16 network');
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const loginExtension = async () => {
+    if (!window.ethereum) {
+      alert('Please connect to Universal Profile Extension');
+      return;
+    }
+
+    try {
+      const { ethereum } = window;
+      if (!verifyChainId()) return;
+      await ethereum
+        .request({
+          method: 'eth_requestAccounts',
+        })
+
+        .then(function (accounts: string[]) {
```

Comment:
you await AND then ?

Why both?

Either do it in await or do it in .then() no?